Sources/HttpPipeline/Response.swift

```diff
@@ -1,7 +1,146 @@
 import Foundation
+import MediaType
+import Prelude
 
 public struct Response {
   public private(set) var status: Status
-  public private(set) var headers: [ResponseHeader]
+  public private(set) var headers: [Header]
   public private(set) var body: Data
+
+  public struct Header {
+    public let name: String
+    public let value: String
+
+    public init(_ name: String, _ value: String) {
+      self.name = name
+      self.value = value
+    }
+
+    public static func allow(_ methods: [Method]) -> Header {
+      return .init("Allow", methods.map(^\.description).joined(separator: ", "))
+    }
+
+    public static func contentLength(_ length: Int) -> Header {
+      return .init("Content-Length", String(length))
+    }
+
+    public static func contentType(_ type: MediaType) -> Header {
+      return .init("Content-Type", type.description)
+    }
+
+    public static func location(_ location: String) -> Header {
+      return .init("Location", location)
+    }
+
+    public static func setCookie(_ name: String, _ value: String, _ options: Set<CookieOption> = [])
+      -> Header {
+
+        let stringValue = ([name + "=" + value] + options.map(^\.description).sorted())
+          .joined(separator: "; ")
+
+        return .init("Set-Cookie", stringValue)
+    }
+
+    public static func wwwAuthenticate(_ authenticate: Authenticate) -> Header {
+      switch authenticate {
+      case let .basic(realm: realm):
+        return .init("WWW-Authenticate", "Basic" + (realm.map { " realm=\"" + $0 + "\"" } ?? ""))
+      }
+    }
+
+    public enum Authenticate {
+      case basic(realm: String?)
+    }
+
+    public enum CookieOption: Hashable, CustomStringConvertible, Comparable {
```

Comment:
do you think it makes sense to convert all the enums to structs?

Reply:
For `Authenticate` maybe, but it's simple enough now while we only support basic auth. For `CookieOption` it's a finite set so `enum` seems to work nicely here. Other thoughts?

Reply:
yeah makes sense. just re-questioning _all_ the enums these days

Reply:
Same! Lol.